docs/tutorial.rst

```diff
@@ -742,50 +742,77 @@ Working with Metadata
+
+Viewing the :class:`tskit.MetadataSchema` for a table can help with understanding
+its metadata as it can contain descriptions and constraints:
 
 .. code-block:: python
 
-    raw_schema = {
-        "codec": "json",
-        "type": "object",
-        "properties": {
-            "accession": {"type": "string"},
-            "pcr": {
-                "name": "PCR Used",
-                "description": "Was PCR used on this sample",
-                "type": "boolean",
-            },
-        },
-        "required": ["accession", "pcr"],
-        "additionalProperties": False,
-    }
-
-The ``codec`` entry in the schema specifies how the metadata should be stored.
-See :ref:`sec_metadata_codecs` for details on the supported codecs. This schema
-defines two properties, both of which are mandatory as they are in the ``required``
-list. To avoid errors we set ``additionalProperties`` to false, as this causes any
-unexpected properties in the metadata to fail validation.
-
-Schemas in tskit are held in :class:`tskit.MetadataSchema`:
+    >>> pprint.pprint(ts.table_metadata_schemas.individual)
+    {'additionalProperties': False,
+     'codec': 'json',
+     'properties': {'accession': {'description': 'ENA accession number',
+                                  'type': 'string'},
+                    'pcr': {'description': 'Was PCR used on this sample',
+                            'name': 'PCR Used',
+                            'type': 'boolean'}},
+     'required': ['accession', 'pcr'],
+     'type': 'object'}
+
+    >>> pprint.pprint(ts.tables.individuals.metadata_schema)
+    (.... as above)
+
+The top-level metadata and schemas for the entire tree sequence is similarly
+accessed with ``ts.metadata`` and ``ts.metadata_schema``.
+
+If there is no schema for a table or top-level metadata, then no decoding is performed
+and ``bytes`` will be returned.
+
+
+++++++++++++++++++++++++++++++
+Modifying metadata and schemas
+++++++++++++++++++++++++++++++
+
+If you are creating or modifying a tree sequence you may want to record or add to the
+metadata. When doing so the schema must be modified first, as it is then used to
+validate and encode the metadata.
+
+Schemas in tskit are held in a :class:`tskit.MetadataSchema`.
+A Python dict representation of the schema is passed to its constructor, which
+will validate the schema:
 
 .. code-block:: python
 
-    schema = tskit.MetadataSchema(raw_schema)
+    schema = tskit.MetadataSchema({"codec": "json" ...... })
 
-Which can be assigned to a table's
+This :class:`tskit.MetadataSchema` can then be assigned to a table or the top-level
+tree sequence e.g.
 :attr:`metadata_schema <tskit.IndividualTable.metadata_schema>`:
 
 .. code-block:: python
 
     tables = tskit.TableCollection(sequence_length=1)
     tables.individuals.metadata_schema = schema
 
-Now that the table has a schema calls to
+This will overwrite any existing schema. Now that the table has a schema calls to
```

Comment:
Perhaps it should be mentioned here that assigning the schema does not validate all pre-existing metadata entries.

Reply:
Note added in https://github.com/tskit-dev/tskit/pull/741/commits/a84fefb9c58431fc6484faa2c4ddbdd113847553